### ISSUE TITLE
Y axis jogs itself on first G0/G1 after power on

### ISSUE THREAD

**Issue by arpreq** [andrivet/ADVi3pp#217]
### Bug Report

- [x] You have checked that you are not creating a **duplicate** of an existing (maybe closed) issue (use the search)
- [x] You have tested with the **latest** stable version of ADVi3++
- [x] You are able to reproduce the bug
- [x] You are able to describe step by step the process to reproduce the bug

#### Description of the bug

* Exact model of your **printer**: Monoprice Maker Select Plus V1
* **ADVi3++** exact version (Mainboard): 4.0.3
* Printing using **SD card** or **USB**: N/A
* **For USB printing only**: Software used to print with its exact version: Octoprint 1.3.11
* **Do you use a BLTouch?**: No
* Expected behaviour: Selected axis is jogged
* Actual behaviour: Selected axis is jogged **AND** Y axis gets jogged in positive direction
* Steps to reproduce:
  - Turn the printer on from a powered-off state
  - Go into "Controls" menu
  - Go into "Move" menu
  - Click the X (either direction), Z (either direction), or Y (negative/top button only) jog buttons
  - Observe the Y axis move in the positive direction in addition to the requested move
    + Even if the Y-negative jog button is pressed, Y will still jog in the positive direction on first press

Notes:
  - Works with Octoprint connected or not
  - Sending equivalent G91+G0/G1 commands through Octoprint terminal exhibits same behavior
  - The unwanted Y axis jog happens simultaneously and has the same feed rate as the requested jog
  - Issue only happens on first axis jog after machine is turned on
  - The home buttons do not trigger the effect
  - Homing the Y axis first before trying a jog results in normal behavior
  - Jogging the Y axis in positive direction (bottom button) first results in normal behavior
  - The extra move does not appear in the Octoprint terminal. Output when moving the X axis on the LCD touch screen:

```
Recv: echo:enqueueing "G91"
Recv: echo:enqueueing "G1 X4 F1000"
Recv: echo:enqueueing "G90"
```

Interestingly, my Z axis does something similar - it lifts about 1cm first when homing any axis for the first time, except for the Z axis.

**Comment by andrivet** [comment]
I can’t reproduce this behavior and you are the only one to report it so far. So I assume it is a problem with your printer, not with the firmware.

**Comment by arpreq** [comment]
Thanks for replying. After a quick search, I found multiple people that mention or ask about this issue in this thread (Gareth, Thomas_Smith, and Aaron) : https://community.advi3pp.com/t/x-and-y-axis-some-times-moves-on-their-own/175/6

I downgraded to 3.0.0 beta3 (which is what I upgraded to 4.0.3 from) and the issue disappeared. Reflashing 4.0.3 brings the issue back.

The behavior really seems to indicate that it might be something in the firmware with how repeatable this is. And the movement doesn't look like it's electronic noise-based, it looks (and feels) like an actual movement command is added in.

EDIT: After even more testing, only firmwares 4.0.2 and 4.0.3 exhibit this problem. Could this be related to your fix for #209?

**Comment by andrivet** [comment]
@arpreq The thread you mentioned is a mix of different problems. Some are however similar to what you see. What you describe is however quite reproducible. It is not the case for other people.
I problem I see in the past similar to what is described in the thread was on the X-axis. It moves alone sometimes. After research and help of some other people, I found it was electronic noise: it triggers the X endstop because of that. The solution was to disable the motors. It was possible in this very particular case.

For Z axis, I am not sure from your description, but I think it is a feature of the firmware to avoid colliding when homing x or y.

For Y axis my guess is that it is a combination of electronic noise and some changes in the firmware between versions 3 and 4. But in fact, I am not sure to fully understand what you described. For example the term "jog" is not clear for me. I am also not sure if your Y axis is moving only 1 step, or always.

Another question: Is it really annoying (if it moves only 1 step)?

**Comment by arpreq** [comment]
> But in fact, I am not sure to fully understand what you described. For example the term "jog" is not clear for me. I am also not sure if your Y axis is moving only 1 step, or always.

Sorry for the confusion - I used the terms "move" and "jog" interchangeably. The Y axis does a full move, as if a G0/G1 command was given that included it.
For example, right after powering on if I send the printer `G0 X50 F1000`, the printer will act exactly as if I sent `G0 X50 Y2.5 F1000`. The Y axis is even left powered on, so the bed can't be moved by hand unless I hit the "motors off" button.

Below is a small video showing the phenomenon. It's not that easy to see, but the bed moves 2.5mm at the same ~~distance and~~ speed as the X axis when I press the X axis jog button.
https://www.dropbox.com/s/1psap8r6f7evh3u/20191009_172555_1.mp4?dl=0

> For Z axis, I am not sure from your description, but I think it is a feature of the firmware to avoid colliding when homing x or y.

You're probably right - the auto Z axis movement might be intended to prevent the nozzle from scraping across the bed as it homes. I'm not super worried about this, but just wanted to mention it because it seemed related.

> Another question: Is it really annoying (if it moves only 1 step)?

Since the Y axis is ~~copying whatever the first move is~~ moving by itself, it's a little dangerous in my opinion. If the Y axis does not have enough room for the move, the bed will crash into the frame of the printer.

EDIT: It turns out that I was mistaken - the unwanted Y axis move is always 2.5mm, but is at the same speed as the requested move. This happens to be the same value added to Y_MIN_POS in cb22aa0 to fix #209.

**Comment by contraspec** [comment]
Hi Sebastien,

im having a similar situation, where if i jog the Z-axis upwards to lift the hot-end, the Y-axis jogs a few mm and stops whilst the Z-axis continues to jog. very strange behavior, which should not happen.

i do believe this could be related to the op's post. 

FW_4.0.3
Di3+mk2
no g-code commands given, no sdcard or usb, only freshly powered on, and jog to lift z axis to clean nozzle.

**Comment by andrivet** [comment]
@contraspec what do you mean by
> i do believe this could be related to the **op's post**

**Comment by Aianir** [comment]
I have the exact same issue as described in this bug report. If I follow the steps to reproduce, the printer performs the same unwanted Y-positive movement.

Exact model of your printer: Wanhao Duplicator i3 Plus
ADVi3++ exact version (Mainboard): 4.0.3
Do you use a BLTouch?: Yes (but not triggered when reproducing the unwanted behaviour)

I also have another issue that I am unable to reproduce; twice after printing, after homing X and Y and is supposed to move the bed to Y200, it moves the X and Y at an incredible speed to about X100 Y100 (maybe the safe home coordinates). If I find any pattern in this I will write up a real bug report, just thought I'd mention it.

**Comment by Salty-Doug** [comment]
Sebastian, 
I can reproduce this behavior repeatedly. If I Stop a print then try to home or move an axis, the stage will move the Y and Z axis until I power the printer off.

**Comment by andrivet** [comment]
@Salty-Doug Are you describing the same behavior that in this issue or another one? Looks like it is another one: the issue steps does not talk about "stop a print". So please fill another issue.
UPDATE: I see that you have in fact created another issue.

**Comment by andrivet** [comment]
@arpreq I am now able to reproduce this bug. Will look at it to fix it.

**Comment by arpreq** [comment]
Great, let me know if you require any more information.

**Comment by Salty-Doug** [comment]
@andrivet I entered this issue as a seperate item #219 
Thanks.

**Comment by andrivet** [comment]
Fixed in 4.0.5
